src/include/migraphx/fp8_ocp_to_nanoo.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#ifndef MIGRAPHX_GUARD_RTGLIB_FP8_OCP_TO_NANOO_HPP
+#define MIGRAPHX_GUARD_RTGLIB_FP8_OCP_TO_NANOO_HPP
+
+#include <migraphx/config.hpp>
+#include <migraphx/pass_manager.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+  
+/**
+ * Convert fp8e4m3fn to fp8e4m3fnuz for hardware that only supports fp8e4m3fnuz data types intrinsically.
+ * Conversion uses the same bit representation and adjusts scaling factors at the dequantization.
+ * Using the same bit representation from fp8e4m3fn to fp8e4m3fnuz halves the floating point representation.
+ * This pass should run before simplify_qdq so that the scales and zero points calculated by simplify_qdq have the correct adjusted scaling factors 
+ */
+struct MIGRAPHX_EXPORT fp8_ocp_to_nanoo
```

Comment:
The pass should be named `fp8_ocp_to_fnuz` or `fp8_to_fnuz`. This matches the way the fp8 type is referred to in migraphx and the way it is referred by [hip](https://rocm.docs.amd.com/projects/HIP/en/latest/reference/fp8_numbers.html), [onnx](https://onnx.ai/onnx/technical/float8.html), [openxla](https://github.com/openxla/stablehlo/blob/main/rfcs/20230321-fp8_fnuz.md), [mlir](https://mlir.llvm.org/docs/Dialects/Builtin/#float8e4m3b11fnuztype), and [pytorch](https://github.com/pytorch/pytorch/pull/115214). Furthermore, a google search for fp8 nano just finds LG earbuds and nothing about fp8 types. So it seems FNUZ is the commonly used term for this type, so we should use `FNUZ` to refer to it in our source code and documentation and not `NANO`.